docs/symb-proto/README.md

```diff
@@ -0,0 +1,98 @@
+Elastic symbolization protocol
+==============================
+
+## `symbfile` format
+
+`symbfile` is our custom file format for efficiently storing large amounts of
+symbol information. A symbfile is a concatenation of length- and message-type
+prefixed protobuf messages. The purpose of a symbfile is to provide a mapping
+from each address in an executable to the corresponding function name, file name
+and line number, including support for inline functions. One or more symbfiles
+always describe just one executable. Recording associations of one or more
+symbfiles with an executable (e.g. via a file hash) is left to other components.
+
+We currently use two different symbol information representations:
+
+- **Range based records ([`RangeV1`])**\
+  These map an address range to symbol information and a `depth` integer that
+  determines the depth within an inline chain. Inline chains are flattened into
+  multiple overlapping range records. To determine the inline trace for any
+  given address, the user would sweep though the whole symbfile and collect all
+  ranges that contain the desired address and then ordering the resulting range
+  records by their `depth` field. This presents the ground truth for symbol
+  information. 
+- **Return pad records ([`ReturnPadV1`])**\
+  These map a single address to the symbols of a full inline trace. We generate
+  such records for each instruction following a `call`. The idea here is that
+  when building stack traces all but the last frame will have addresses that
+  point to such return addresses. Special casing these allows the symbolization
+  service to proactively insert all symbols for all non-leaf frames which then
+  massively reduces the amount of frames that need to be symbolized lazily.
+
+While the symbfile format would generally also allow mixing both record types
+into a single file, we currently always generate a separate symbfile per record
+kind. 
+
+More details about the format itself can be found in the documentation comments
+of the [protobuf definition][symbfile-proto].
+
+[`RangeV1`]: ./symbfile.proto#L120
+[`ReturnPadV1`]: ./symbfile.proto#L212
+[symbfile-proto]: ./symbfile.proto
```

Comment:
No: the link on line 37 refers to this "destination definition" (not sure what the proper Markdown terminology is here, but essentially it provides an alias for a URL). Arguably this link is so short that it could be inlined now: in the monorepo this used to be a long permalink, hence pulling it out.